mysql-test/suite/sys_vars/t/log_query_errors_basic.test

```diff
@@ -0,0 +1,159 @@
+SET @old_log_query_errors = @@global.log_query_errors;
+
+--echo # check default empty value
+SELECT @@global.log_query_errors;
+SELECT @@session.log_query_errors;
+SHOW GLOBAL VARIABLES LIKE 'log_query_errors';
+SHOW SESSION VARIABLES LIKE 'log_query_errors';
+--disable_warnings
+SELECT * FROM performance_schema.global_variables WHERE variable_name='log_query_errors';
+SELECT * FROM performance_schema.session_variables WHERE variable_name='log_query_errors';
+--enable_warnings
+
+--echo
+--echo # set global to ALL, session is unset
+SET GLOBAL log_query_errors=all;
+
+SELECT @@global.log_query_errors;
+SELECT @@session.log_query_errors;
+SHOW GLOBAL VARIABLES LIKE 'log_query_errors';
+SHOW SESSION VARIABLES LIKE 'log_query_errors';
+--disable_warnings
+SELECT * FROM performance_schema.global_variables WHERE variable_name='log_query_errors';
+SELECT * FROM performance_schema.session_variables WHERE variable_name='log_query_errors';
+--enable_warnings
+
+--echo
+--echo # unset global, set session to ALL
+SET GLOBAL log_query_errors="";
+SET SESSION log_query_errors=all;
+
+SELECT @@global.log_query_errors;
+SELECT @@session.log_query_errors;
+SHOW GLOBAL VARIABLES LIKE 'log_query_errors';
+SHOW SESSION VARIABLES LIKE 'log_query_errors';
+--disable_warnings
+SELECT * FROM performance_schema.global_variables WHERE variable_name='log_query_errors';
+SELECT * FROM performance_schema.session_variables WHERE variable_name='log_query_errors';
+--enable_warnings
+
+--echo
+--echo # set both to ALL
+SET GLOBAL log_query_errors=all;
+SET SESSION log_query_errors=all;
+
+SELECT @@global.log_query_errors;
+SELECT @@session.log_query_errors;
+SHOW GLOBAL VARIABLES LIKE 'log_query_errors';
+SHOW SESSION VARIABLES LIKE 'log_query_errors';
+--disable_warnings
+SELECT * FROM performance_schema.global_variables WHERE variable_name='log_query_errors';
+SELECT * FROM performance_schema.session_variables WHERE variable_name='log_query_errors';
+--enable_warnings
+
+--echo
+--echo # set to default
+SET SESSION log_query_errors=default;
+SET GLOBAL log_query_errors=default;
+SELECT @@global.log_query_errors;
+SELECT @@session.log_query_errors;
+
+--echo
+--echo # set some codes
+SET GLOBAL log_query_errors="1010,1011,1012";
+SET SESSION log_query_errors="2010,2011,2012";
```

Comment:
1. Do we need to treat them as string? ie, will it work without the double or single quotes?
2. Also can you try with the longest number possible? 2^64
3. Does it check if an error code is a valid or not? ie if an error code is > the "max allowed errorcode " or not? etc

Reply:
> Do we need to treat them as string? ie, will it work without the double or single quotes?

Yes, it should be a string

> Also can you try with the longest number possible? 2^64

There was no check for biggest acceptable number. I've added one.

> Does it check if an error code is a valid or not? ie if an error code is > the "max allowed errorcode " or not? 

No, at the moment it doesn't know about existing error codes.

> For a single error number, is it possible to do this SET GLOBAL log_query_errors=1010?

No, it still should be a string - "1010"